tests/test_data_structures.cpp

```diff
@@ -826,4 +826,80 @@ TEST_CASE( "syncDiagonalOp", "[data_structures]" ) {
         destroyQureg(qureg, QUEST_ENV);
         destroyDiagonalOp(op, QUEST_ENV);
     }
-}
\ No newline at end of file
+}
+
+
+
+/** @sa initDiagonalOpFromPauliHamil
+ * @ingroup unittest
+ * @author Tyson Jones, Milos Prokop
+ */
+TEST_CASE( "DiagonalOpFromPauliHamil", "[data_structures]" ) {
+
+    // must be at least one amplitude per node
+    int minNumQb = calcLog2(QUEST_ENV.numRanks);
+    if (minNumQb == 0)
+        minNumQb = 1;
+
+    // try 10 valid number of qubits (even for validation)
+    int numQb = GENERATE_COPY( range(minNumQb+1, minNumQb+10) );
+    int numSumTerms = GENERATE_COPY( range(5, 30) );
+
+    DiagonalOp op = createDiagonalOp(numQb, QUEST_ENV);
+    PauliHamil hamil = createPauliHamil(numQb, numSumTerms);
+
+    setRandomPauliSum(hamil, true);
+
+    qreal* zero_imag = (qreal*) malloc((1<<numQb)*sizeof(qreal));
+    for(size_t i = 0; i < (1<<numQb); ++i)
+    	zero_imag[i] = 0.;
+
+    SECTION( "correctness" ) {
+
+    	QMatrix expected_matrix = toQMatrix(hamil);
+    	REQUIRE(isDiagonal(expected_matrix));
```

Comment:
'meta unit test' verifications like this should *not* exist in the unit test for an API function, which should be reserved for testing the API function (not test utilities). They should be inside the test utility itself